cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -270,7 +271,12 @@ func gitIndex(c gitIndexConfig, o *indexArgs) error {
 		}
 	}
 
-	debug.Printf("successfully fetched git data for %q (%d commit(s)) in %s", o.Name, successfullyFetchedCommitsCount, fetchDuration)
+	logger.Debug("successfully fetched git data",
```

Comment:
As of now the new logger will not be able to use the `debug` flag for zoekt-sourcegraph-indexserver.

The current debug log writer will use the debug flag to determine whether it should output messages. If the debug flag is not passed then the environment variable `SRC_LOG_LEVEL` is used to determine this.

The new logger always uses that environment variable to determine whether to write debug log message. The check is made in [Init()](https://sourcegraph.com/github.com/sourcegraph/log/-/blob/init.go?L49) and then passed to globallogger.go. After init we can only increase the level of the logger, not decrease.  See [IncreaseLevel()](https://sourcegraph.com/github.com/sourcegraph/log/-/blob/logger.go?L72). E.g. we cannot later change the init'd root logger from, say, Info level to Debug level (confirmed in my local testing), but we can change from Debug Level to Info Level.

If we are okay with relying on env var and not debug flag for the new message log then we can proceed with this current code change. Alternatively we can make this an info message so that this fetched git data duration is always logged.

Reply:
you can also [os.Setenv](https://pkg.go.dev/os#Setenv) if debug is set, before log.Init